### PR TITLE
Node api

### PR DESCRIPTION
**Pull request by allouis**

### CODE REVIEW

File: api/bin/setup-db [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    db.put('mention:' + mention.id, mention, error => {
+      if (error) return reject(error)
+      resolve()
+    })
+  })
+}
+
+const puts = mentions.map(addMention)
+
+Promise.all(puts).then(function(){
```
Comment (larister) [inline api/bin/setup-db]: y u no arrow here just out of interest?

File: api/index.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const debug = require('debug')('server')
+debug('Booting API')
+
+const url = require('url')
+const server = require('http').createServer()
+const router = require('./router')
+
+server.on('request', (req, res) => {
+  debug('request %s %s', req.method, req.url)
```
Comment (larister) [inline api/index.js]: Small thing but string interpolation instead here?

File: api/package.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "debug": "^2.2.0",
+    "json-object-stream": "^0.1.1",
+    "level": "^1.3.0",
+    "routes": "^2.0.0"
+  },
+  "devDependencies": {
+    "standard": "^5.3.1"
```
Comment (larister) [inline api/package.json]: Controversial, happy to give it a go though :+1:

File: api/package.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    "test": "standard",
+    "start": "DEBUG=server,db,router node index.js"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "debug": "^2.2.0",
+    "json-object-stream": "^0.1.1",
+    "level": "^1.3.0",
+    "routes": "^2.0.0"
```
Comment (larister) [inline api/package.json]: NABD but this is no longer under active development https://www.npmjs.com/package/routes#alternative-routers

File: api/package.json (hunk 3 above)
Comment (jdelight) [inline api/package.json]: ^^ likewise - interesting! definitely leaning away from the "all-the-lint-files" approach personally.

File: api/router.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+router.addRoute('/api/status', (req, res) => {
+  res.writeHead(200)
+  res.end()
+})
+
+router.addRoute('/api/mentions', (req, res) => {
+  const delay = +req.params.delay || 0
+  debug('delay of %sms', delay)
+  setTimeout(_ => db.getMentions().pipe(res), delay)
```
Comment (larister) [inline api/router.js]: Can we add a debug statement for the return as well?

**Comment by jdelight** [comment]
@allouis LGTM :+1:

File: api/bin/setup-db (hunk 1 above)
Comment (allouis) [inline api/bin/setup-db]: forgot! will add :relaxed:

File: api/index.js (hunk 2 above)
Comment (allouis) [inline api/index.js]: Fair point

File: api/package.json (hunk 3 above)
Comment (allouis) [inline api/package.json]: Yeah, I didn't like the idea of it, but took about a day or two to get used to it, it's so much easier to just get started and not arguing over each little setting

File: api/package.json (hunk 4 above)
Comment (allouis) [inline api/package.json]: It's stable, i think we're fine :)

File: api/router.js (hunk 5 above)
Comment (allouis) [inline api/router.js]: Sure

File: api/package.json (hunk 4 above)
Comment (larister) [inline api/package.json]: Sounds good, it's hardly rocket science anyway :D

File: api/package.json (hunk 3 above)
Comment (larister) [inline api/package.json]: :+1: Just don't tell crockford

**Comment by larister** [comment]
LGTM, wanna merge @jdelight?

**Comment by jdelight** [comment]
One minor thing @allouis would it be possible to set the response 'Content-Type' header to `application/json`? Makes formatting much better with tools like httpie and postman. Then :train: to Mergetown.

**Comment by larister** [comment]
Ah yeah good catch

**Pull request merged**